src/securejoin.rs

```diff
@@ -609,6 +609,8 @@ pub(crate) async fn observe_securejoin_on_other_device(
                 peerstate.set_verified(PeerstateKeyType::GossipKey, fingerprint, addr)?;
                 peerstate.prefer_encrypt = EncryptPreference::Mutual;
                 peerstate.save_to_db(&context.sql).await.unwrap_or_default();
+
+                ChatId::set_protection_for_contact(context, contact_id).await?;
```

Comment:
This change is not related to https://github.com/deltachat/securejoin/blob/main/uxscenarios/resetup.md#i-happy-path-and-resetup-scenarios-with-green-checkmarked-chatting, but to fix another potential problem I noticed: When observing a securejoin on another device, the 1:1 chat should also be marked as verified.

I didn't manage to write a test that fails without this line here, so I don't know whether it is actually needed. I think that it should be fine to just merge it like this; otherwise, at least until next Wednesday, I won't have the time to dig further into this. An alternative would be to merge the PR without this addition.